### PR TITLE
[BS-351][TW] Updating errors and adding missing sections

### PR DESCRIPTION
**Pull request by glidester**

### CODE REVIEW

File: public/api/conf/1.0/application.raml [hunk 1]
```diff
@@ -257,18 +286,6 @@ documentation:
                     description: One or more values cannot validated for the Json item
```
Comment (udinobi) [inline public/api/conf/1.0/application.raml]: `cannot **be** validated` ?

File: public/api/conf/1.0/docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+You can use the sandbox environment to [test this API](https://developer.service.hmrc.gov.uk/api-documentation/docs/testing).
+
+It currently only supports [stateless behaviour](https://developer.service.hmrc.gov.uk/api-documentation/docs/testing/stateful-behaviour).
```
Comment (udinobi) [inline public/api/conf/1.0/docs/testing.md:3]: "stateless behaviour" and "stateful-behaviour"
Maybe something wrong ?

Comment (glidester) [reply public/api/conf/1.0/docs/testing.md:3]: No, that page covers both scenarios: https://developer.service.hmrc.gov.uk/api-documentation/docs/testing/test-users-test-data-stateful-behaviour

File: public/api/conf/1.0/application.raml (hunk 1 above)
Comment (glidester) [reply public/api/conf/1.0/application.raml]: Done

File: public/api/conf/1.0/application.raml [hunk 3]
```diff
@@ -349,30 +366,34 @@ documentation:
...
-                  description: Invalid Date
+                invalidJson:
+                  description: Invalid Json
                   value:
-                    code: INVALID_DATE
-                invalidDateRange:
-                  description: Invalid Date Range
+                    code: INVALID_JSON
+                invalidJsonItem:
+                  description: One or more values cannot validated for the Json item
```
Comment (udinobi) [inline public/api/conf/1.0/application.raml]: `cannot be validated` even here ?

Comment (glidester) [reply public/api/conf/1.0/application.raml]: Done

**Review by udinobi — APPROVED** [review]

**Pull request merged**